AT/hub_thread/hub_routine.py

```diff
@@ -0,0 +1,80 @@
+    while not first_status_has_arrived:
+        if not sim_to_hub.empty():
+            # Parse the simulator message
+            signal = sim_to_hub.get()
+            if signal['type'] == 'window':
+                # Parse the signal
+                tf_window = signal['content']
+
+                # Retrieve the telemetry feed variables and send an initialization command to the frontend
+                tf_variables = list(tf_window['values'].columns.values)
+                tf_variables.remove('timestamp')
+                command = {'type': 'initialize_telemetry', 'variables': tf_variables}
+                async_to_sync(channel.send)(channel_name, command)
+
+                # Put the first simulator output back in the queue and update the while loop condition
+                sim_to_hub.put(signal)
+                first_status_has_arrived = True
+        else:
+            time.sleep(0.01)
+
+    # Set and initialize the ping routine counters
+    life_limit = 40
+    time_since_last_ping = 0
+    timer_start = time.time()
+
+    # Start the hub routine
+    while time_since_last_ping < life_limit:
+        # Check the frontend input queue
+        if not front_to_hub.empty():
+            signal = front_to_hub.get()
+            print('New message from the frontend: ' + signal)
+            if signal == 'stop':
+                break
+            elif signal == 'ping':
+                time_since_last_ping = 0
+                timer_start = time.time()
+
+        # Check the simulator input queue
+        if not sim_to_hub.empty():
+            signal = sim_to_hub.get()
+            if signal['type'] == 'window':
+                # Parse the signal
+                tf_window = signal['content']
+
+                # Send the telemetry feed window to the anomaly detection queue
+                last_window = {'type': 'window', 'content': tf_window}
+                hub_to_ad.put(last_window)
+
+                # Update and send the telemetry update command for the frontend
+                command = {'type': 'telemetry_update',
+                           'values': tf_window['values'].to_json(),
+                           'info': tf_window['info'].to_json()}
+                async_to_sync(channel.send)(channel_name, command)
+
+        # Check the diagnosis output queue
+        if not diag_to_hub.empty():
+            signal = diag_to_hub.get()
+            if signal['type'] == 'diagnosed_anomalies':
+                diagnosed_anomalies = signal['content']
+                command = {'type': 'at_analysis', 'diagnosed_anomalies': diagnosed_anomalies}
+                async_to_sync(channel.send)(channel_name, command)
+
+        # Update while loop condition and wait
+        time_since_last_ping = time.time() - timer_start
+        time.sleep(0.1)
+
+    hub_to_sim.put({'type': 'stop', 'content': ''})
+    hub_to_ad.put({'type': 'stop', 'content': ''})
+
+    print('Hub thread stopped.')
+    return None
```

Comment:
return None is implicit, no need to put it, also, last line should always be empty